config.go

```diff
@@ -736,14 +739,14 @@ func loadConfig() (*config, []string, error) {
 	// This allows .onion address traffic to be routed through a different
 	// proxy than normal traffic.
 	if cfg.OnionProxy != "" {
-		cfg.oniondial = func(a, b string) (net.Conn, error) {
-			proxy := &socks.Proxy{
-				Addr:     cfg.OnionProxy,
-				Username: cfg.OnionProxyUser,
-				Password: cfg.OnionProxyPass,
-			}
-			return proxy.Dial(a, b)
+		onionProxy, err := socks5.New(cfg.OnionProxy)
```

Comment:
I'd prefer the options to be included in the call to `New`, like the previous package.  It's easy to forget to set additional options when they're aren't part of the `New` call struct params.